lib/engine/part/city.rb

```diff
@@ -17,6 +17,7 @@ def initialize(revenue, slots = 1)
         @slots = slots.to_i
         @tokens = Array.new(@slots)
         @reservations = []
+        @connected_edges = {}
```

Comment:
intsead of doing this, let's add to part base an attr called paths

this way every part has a back reference to it's paths

you may need to grep the code a bit to refactor this

Reply:
Given that when the parts are created, all the paths are not necessarily created (https://github.com/tobymao/18xx/blob/3d43922ace1febb16591b609272a3cd2ea36d036/lib/engine/tile.rb#L68-L104), would you be fine with the different part classes having a `paths` method that calculates `@paths` based on `@tile`?

Reply:
For example, here's how `Engine::Part::City` could look:

```
def initialize
  # ...
  @paths = {}
end

# ...

def paths
  @paths[tile.rotation] ||= tile.paths.select { |p| p.city == self }
end

def connected_edges
  paths.flat_map(&:exits).sort
end